### PR TITLE
Avoid using h24 in examples

### PR DESCRIPTION
**Pull request by hsivonen**
### Description

Avoid suggesting the use of "h24" anywhere.

### Motivation

The hour cycle h24 is specified for completeness, as its existence in the mathematical sense is implied by h11, h12, and h23. However, h24 isn't used anywhere in the world (as far as CLDR is aware).

Superficially, h24 looks like the designator for "the 24-hour clock", but the correct designator for "the 24-hour clock" is h23.

### Additional details

See https://github.com/unicode-org/cldr-json/blob/80a94b0f6c3a34d6e2dc0dca8639a54babc87f94/cldr-json/cldr-core/supplemental/timeData.json (h means h12, H means h23, K means h11. h24 would be k, which doesn't appear there.)

### Related issues and pull requests

https://github.com/mdn/content/pull/28701

### CODE REVIEW

File: live-examples/js-examples/intl/intl-locale.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const korean = new Intl.Locale('ko', {
   script: 'Kore',
   region: 'KR',
-  hourCycle: 'h24',
+  hourCycle: 'h12',
```
Comment (Josh-Cena) [inline live-examples/js-examples/intl/intl-locale.js]: Maybe use `h23`? Otherwise we have two `h12`s.

Comment (hsivonen) [reply live-examples/js-examples/intl/intl-locale.js]: Done.

**Review by Josh-Cena — APPROVED** [review]
LGTM; @bsmth could you merge? I used to have the perms but not anymore.

**Comment by bsmth** [comment]
Thanks both, merging shortly

**Pull request merged**